pkg/blockstorage/awsebs/awsebs.go

```diff
@@ -238,14 +239,15 @@ func (s *EbsStorage) snapshotParse(ctx context.Context, snap *ec2.Snapshot) *blo
 		Type: s.Type(),
 		ID:   aws.StringValue(snap.VolumeId),
 	}
+
 	// TODO: fix getting region from zone
 	return &blockstorage.Snapshot{
 		ID:           aws.StringValue(snap.SnapshotId),
 		Tags:         tags,
 		Type:         s.Type(),
 		Encrypted:    aws.BoolValue(snap.Encrypted),
 		SizeInBytes:  aws.Int64Value(snap.VolumeSize) * blockstorage.BytesInGi,
-		Region:       aws.StringValue(s.config.Region),
+		Region:       aws.StringValue(s.Ec2Cli.EC2API.(*ec2.EC2).Config.Region),
```

Comment:
```suggestion
		Region:       aws.StringValue(s.Ec2Cli.Config.Region),
```
Wouldn't this work? Referring from the original changes from https://github.com/kanisterio/kanister/pull/3549/files

Reply:
Its not a best practice to type assert interfaces directly. It might cause panic.
```
    if s, ok := s.Ec2Cli.EC2API.(*ec2.EC2); ok {
        region = s.Ec2Cli.EC2API.(*ec2.EC2).Config.Region
    } else {
        // handle if assertion not possible
    }
```
We can first check if its assertion is possible. But overall I think we should look for better way. Can we not revert it to the original way of handling like in https://github.com/kanisterio/kanister/pull/3549/files?

Reply:
Can we please revert back the interface change? We need to find better way to do this.

Reply:
have revert back the interface changes.